### PR TITLE
[SPARK-30361][REST] Monitoring URL do not redact information about environment

### PR DESCRIPTION
**Pull request by ajithme**
UI and event logs redact sensitive information. But the monitoring URL, https://spark.apache.org/docs/latest/monitoring.html#rest-api , specifically /applications/[app-id]/environment does not, which is a security issue.

### What changes were proposed in this pull request?
REST api response is redacted before sending it

### Why are the changes needed?
If no redaction is done for rest API call, it can leak security information

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Tested manually

### CODE REVIEW

**Comment by ajithme** [comment]
cc @dongjoon-hyun @srowen

File: core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala [hunk 1]
```diff
@@ -97,7 +99,15 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
...
   @Path("environment")
-  def environmentInfo(): ApplicationEnvironmentInfo = withUI(_.store.environmentInfo())
+  def environmentInfo(): ApplicationEnvironmentInfo = withUI { ui =>
+    val envInfo = ui.store.environmentInfo()
+    new v1.ApplicationEnvironmentInfo(
+      envInfo.runtime,
+      Utils.redact(ui.conf, envInfo.sparkProperties),
+      Utils.redact(ui.conf, envInfo.hadoopProperties),
+      Utils.redact(ui.conf, envInfo.systemProperties),
+      Utils.redact(ui.conf, envInfo.classpathEntries))
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala]: As in the other PR, I don't know if we need to redact the classpath?

**Review by srowen — COMMENTED** [review]
Seems plausible but @vanzin is probably a better person to check with

File: core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala (hunk 1 above)
Comment (HeartSaVioR) [reply core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala]: +1. I'm not sure what's the point of applying redact to classpath entries, and why exposing classpath entries without redacting is considered as security issue. Moreover I'm not sure what would be matched to redact patterns from classpath entries.

Comment (ajithme) [reply core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala]: jar name containing key words like password, token etc.. i have updated the PR to skip redacting classpath.

**Review by srowen — APPROVED** [review]

**Comment by ajithme** [comment]
retest this please

> **[Test build #4978 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/4978/testReport)** for PR 27018 at commit [`6a9ea54`](https://github.com/apache/spark/commit/6a9ea54f7dc54c6dacb5b365f6c76812df556a5b).
> 
> * This patch **fails Spark unit tests**.
> * This patch merges cleanly.
> * This patch adds no public classes.

test failures do not look related to this PR

**Comment by ajithme** [comment]
retest this please

**Comment by srowen** [comment]
Jenkins retest this please

**Review by HeartSaVioR — APPROVED** [review]
LGTM

**Comment by ajithme** [comment]
Gentle ping @vanzin @srowen @HyukjinKwon

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**